Authentication/AuthenticationComponentsFactory.swift

```diff
@@ -172,18 +187,30 @@ public extension LoginComponentsFactory {
         return LoginCredentialsValidator()
     }
     
-    public func createLoginViewModel<SessionService: SessionServiceType>(withSessionService sessionService: SessionService, credentialsValidator: LoginCredentialsValidator) -> LoginViewModelType {
-        return LoginViewModel(sessionService: sessionService, credentialsValidator: createLogInCredentialsValidator())
+    public func createLoginViewModelFactory<SessionService: SessionServiceType>(
+        withSessionService sessionService: SessionService,
+        credentialsValidator: LoginCredentialsValidator) -> ([LoginProvider]) -> LoginViewModelType {
+        return { (loginProviders: [LoginProvider]) in
+            for provider in loginProviders {
+                provider.configure()
```

Comment:
This seems a little bit strange. What is it for? The strange thing for me is that this is call when the login view model is created, and they may not be configured when the login view receives them, if we call createLoginView first. If they need a configuration, I guess the one intending to use it should do it before giving it to us in the framework, since that configuration doesn't need anything from us.